src/main/java/com/palantir/gradle/gitversion/Git.java

```diff
@@ -166,30 +160,15 @@ public Boolean isClean() {
 
     public String describe(String prefix) {
         try {
-            // Get SHAs of all tags, we only need to search for these later on
-            Set<String> tagRefs = new HashSet<>();
-            for (String tag : LINE_SPLITTER.splitToList(runGitCmd("show-ref", "--tags", "-d"))) {
-                List<String> parts = WORD_SPLITTER.splitToList(tag);
-                Preconditions.checkArgument(parts.size() == 2, "Could not parse output of `git show-ref`: %s", parts);
-                tagRefs.add(parts.get(0));
-            }
-
-            List<String> revs = LINE_SPLITTER.splitToList(runGitCmd("rev-list", "--first-parent", "HEAD"));
-            for (int depth = 0; depth < revs.size(); depth++) {
-                String rev = revs.get(depth);
-                if (tagRefs.contains(rev)) {
-                    String exactTag = runGitCmd("describe", "--tags", "--exact-match", "--match=" + prefix + "*", rev);
-                    if (!exactTag.isEmpty()) {
-                        return depth == 0
-                                ? exactTag
-                                : String.format("%s-%s-g%s", exactTag, depth, GitUtils.abbrevHash(revs.get(0)));
-                    }
-                }
-            }
-
-            // No tags found, so return commit hash of HEAD
-            return GitUtils.abbrevHash(runGitCmd("rev-parse", "HEAD"));
-        } catch (IOException | InterruptedException | RuntimeException e) {
+            return runGitCommand(
+                    "describe",
+                    "--tags",
+                    "--always",
+                    "--first-parent",
+                    "--abbrev=" + SHA_ABBR_LENGTH,
```

Comment:
This is only used in once place, let's just inline it and avoid the string concatenation

```suggestion
                    "--abbrev=7",
```